components/brave_shields/resources/panel/state/component_types.ts

```diff
@@ -2,9 +2,38 @@
 // This Source Code Form is subject to the terms of the Mozilla Public
 // License, v. 2.0. If a copy of the MPL was not distributed with this file,
 // you can obtain one at https://mozilla.org/MPL/2.0/.
+
+import { Url } from 'gen/url/mojom/url.mojom.m.js'
+
 export enum ViewType {
   ScriptsList,
   AdsList,
   HttpsList,
   Main
 }
+
+export enum ResourceType {
+  Script,
+  Ad,
+  Http
+}
+
+export enum ResourceState {
+  Blocked,
+  AllowedOnce
+}
+
+export type ResourceInfo = {
+  url: Url
+  state: ResourceState
+  type: ResourceType
+}
+
+export function makeResourceInfoList (data: Url[],
```

Comment:
This is a helper function and shouldn't be located in the "component types" section as it's not a component type. Please apply the transformation in the hooks file I mentioned earlier, and remove this function from this location.

Reply:
removed it